packages/minter-contracts/ligo/src/fixed_price_sale/README.md

```diff
@@ -0,0 +1,296 @@
+
+* [fa2_interface.ts](../../../src/fa2-interface.ts) line 46.
+
+
+These above steps outline apply irrespective of which version of the fixed price
+sale contract is in use. The primary difference is the record type of the
+variant associated with this entrypoint. For [`SALE_FA2`](#sale-fa2-section) we
+have the record type `init_sale_param` we have:
+
+
+``` ocaml
+type init_sale_param =
+[@layout:comb]
+{
+  sale_price: nat;
+  sale_tokens_param: sale_tokens_param;
+}
+```
+
+This record contains the sale price of the NFT in positive integer units
+corresponding to the FA2 token used to price the NFT along with
+`sale_tokens_param` field described earlier in this document (the section
+[Storage](#storage-section)). We have something similar for
+[`SALE_TEZ`](#sale-tez-section):
+
+``` ocaml
+type init_sale_param_tez =
+[@layout:comb]
+{
+  sale_price: tez;
+  sale_token_param_tez: sale_token_param_tez;
+}
+```
+
+This record, like the one above, contains the sale price in tez (instead of some
+FA2 token) and the `sale_token_param_tez`, again described earlier in this
+document (the section [Storage](#torage-section)).
+
+### %buy
+
+Like in the previous section, an administrator can prevent this entrypoint's
+invocation by pausing the contract. Nonetheless, under normal operation, this
+entrypoint will allow a potential buyer to claim the NFT for sale provided they
+have funds. If they do not have the funds, they will not lay claim to the NFT
+(it will remain under the ownership of the seller). If someone else has bought
+the NFT or if the seller has cancelled the sale, the entrypoint will emit an
+error message of "NO_SALE".
+
+The record associated to the `Buy` variant is either `sale_param` or
+`sale_param_tez`. Earlier sections of this document (see
+[Storage](#storage-section)) contain their definitions.
+
+More specifically, let's go through the steps that would transpire when the
+`%buy` entrypoint is invoked. As stated shortly before, the `Buy` variant either
+contains the `sale_param` or `sale_param_tez` record depending upon the version
+of the fixed price sale contract. Regardless of which, here are the basic steps
+that should occur following this entrypoint's invocation.
+
+1. Determine if the sale specified in record found in the `Buy` variant is a
+   currently active sale. This amounts to a big_map lookup. If this lookup
+   retrieval is successful, continue to the following steps while keeping track
+   of the price attached to the sale's contents, otherwise emit the
+   `failwith` operation with the following message "NO SALE".
+2. Create a set of operations corresponding to the transfer of the sale price
+   (recorded in the last step) from the buyer's available funds. Of course, this
+   will fail if the required funds are not available in the buyer's "purse". To
+   further clarify, when buying NFTs with FA2 tokens, this set of operations
+   includes two operations. Namely, the first operation sends FA2 tokens to the
```

Comment:
If FA2 tokens are to be transferred from buyer to the marketplace contract by marketplace contract itself, buyer needs to add market place as an FA2 token operator.